lib/libc/newlib/libc-hooks.c

```diff
@@ -291,6 +292,18 @@ void *_sbrk(intptr_t count)
 }
 __weak FUNC_ALIAS(_sbrk, sbrk, void *);
 
+static LIBC_DATA SYS_MUTEX_DEFINE(heap_mutex);
```

Comment:
When CONFIG_USERSPACE=n, making this a spinlock would be much smaller/faster.  But they I guess someday we'll get around to implementing a futex backend for sys_mutex which would have the same property.